src/v/raft/heartbeat_manager.cc

```diff
@@ -72,84 +63,45 @@ heartbeat_manager::follower_request_meta::~follower_request_meta() noexcept {
     }
 }
 
-static heartbeat_requests requests_for_range(
-  const consensus_set& c, clock_type::duration heartbeat_interval) {
+heartbeat_manager::heartbeat_requests heartbeat_manager::requests_for_range() {
     absl::node_hash_map<
       model::node_id,
       ss::chunked_fifo<std::pair<
         heartbeat_metadata,
         heartbeat_manager::follower_request_meta>>>
       pending_beats;
-    if (c.empty()) {
+
+    if (_consensus_groups.empty()) {
         return {};
     }
 
     // Set of follower nodes whose heartbeat_failed status indicates
     // that we should tear down their TCP connection before next heartbeat
     absl::flat_hash_set<model::node_id> reconnect_nodes;
```

Comment:
don't you have to size it. : `reconnect_nodes.resize( groups.size() )`

Reply:
the size of `reconnect_nodes` will be mostly empty as the set contains nodes which connection should be reset.